py-mupdf: new port (PyMuPDF)

markmentovai opened the pull request:
#### Description

<!-- Note: it is best to make pull requests from a branch rather than from master -->

###### Type(s)
<!-- update (title contains ": U(u)pdate to"), submission (new Portfile) and CVE Identifiers are auto-detected, replace [ ] with [x] to select -->

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on
<!-- Triple-click and copy the next line and paste it into your shell. It will copy your OS and Xcode version to the clipboard. Paste it here replacing this section.
sh -c 'echo "macOS $(sw_vers -productVersion) $(sw_vers -buildVersion) $(uname -m)"; xcode=$(xcodebuild -version 2>/dev/null); if [ $? == 0 ]; then echo "$(echo "$xcode" | awk '\''NR==1{x=$0}END{print x" "$NF}'\'')"; else echo "Command Line Tools $(pkgutil --pkg-info=com.apple.pkg.CLTools_Executables | awk '\''/version:/ {print $2}'\'')"; fi' | tee /dev/tty | pbcopy
-->
macOS 13.3 22E252 arm64
Xcode 14.2 14C18

###### Verification <!-- (delete not applicable items) -->
Have you

- [ ] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [ ] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [ ] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? <!-- Please don't open a new Trac ticket if you are submitting a pull request. -->
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

<!-- Use "skip notification" (surrounded with []) to avoid notifying maintainers -->

reneeotten commented:
this should replace then #15971?

reneeotten commented on python/py-mupdf/Portfile:
```diff
@@ -0,0 +1,47 @@
+python.pep517       yes
+
+if {${name} ne ${subport}} {
+    livecheck.type  none
```
this is the default set by the python PG and should thus be removed

reneeotten commented on python/py-mupdf/Portfile:
```diff
@@ -0,0 +1,47 @@
+
+revision            0
+categories          python graphics
+platforms           darwin
```
this is the default and can therefore be removed

reneeotten commented on python/py-mupdf/Portfile:
```diff
@@ -0,0 +1,47 @@
+    livecheck.type  none
+
+    depends_build   port:swig-python
+    depends_lib     port:mupdf
```
please change both specification to depends_*-append as not to overwrite dependencies added by the python PG

markmentovai commented:
https://github.com/macports/macports-ports/pull/18164#issuecomment-1493453663:
> this should replace then #15971?

Yes.

I don’t love doing that, but #15971 seems to have lost traction with no activity in 7 months, and by now, the version of PyMuPDF that it seeks to add is already out of date.

I addressed all review feedback. Thanks, Renee.

reneeotten commented on python/py-mupdf/Portfile:
```diff
@@ -0,0 +1,44 @@
+MuPDF can access files in PDF, XPS, OpenXPS, CBZ, EPUB and FB2 (eBooks) formats, and it is known for its top performance and exceptional rendering quality. \
+\
+With PyMuPDF you can access files with extensions like .pdf, .xps, .oxps, .cbz, .fb2 or .epub. In addition, about 10 popular image formats can also be handled like documents: .png, .jpg, .bmp, .tiff, etc.
+
```
please do "proper" line continuations for the long description so that lines in the Portfiles don't extend beyond 80 characters.

reneeotten commented on python/py-mupdf/Portfile:
```diff
@@ -0,0 +1,44 @@
+
+if {${name} ne ${subport}} {
+    depends_build-append port:swig-python
+    depends_lib-append port:mupdf
```
please make sure that the "four spaces" indentation is correct in the lines above
